### ISSUE TITLE
Can't create instance template using gce-container

### ISSUE THREAD

**Issue by T0shik** [terraform-google-modules/terraform-google-container-vm#80]
### TL;DR

template with source image of gce-container isn't configured to run specified container.

### Expected behavior

After running `terraform apply` I'd expect the Deploy Container options in GCP UI to be filled out with those of `gce-container`. And when creating an instance or instance group using the template, the instance would run the specified container in `gce-container`.

### Observed behavior

template is not configured to run a container, only to use container optimized OS. Instances created with the template do not actually run the `nginxdemos/hello` container.

### Terraform Configuration

```hcl
provider "google" {
  project = "proj1"
  region  = "us-east1"
  zone    = "us-east1-d"
}

provider "google-beta" {
  project = "proj1"
  region  = "us-east1"
  zone    = "us-east1-d"
}

data "google_compute_default_service_account" "default_service_account" {
}

module "gce-container" {
  source = "github.com/terraform-google-modules/terraform-google-container-vm"

  container = {
    image = "nginxdemos/hello"
    env = [
      {
        name  = "EXAMPLE"
        value = "Hello World"
      }
    ]
  }
}

module "mig_template" {
  source       = "terraform-google-modules/vm/google//modules/instance_template"
  version      = "~> 6.0"
  name_prefix  = "example"
  machine_type = "n1-standard-2"
  service_account = {
    email = data.google_compute_default_service_account.default_service_account.email
    scopes = [
      "https://www.googleapis.com/auth/datastore",
      "https://www.googleapis.com/auth/pubsub",
      "https://www.googleapis.com/auth/servicecontrol",
      "https://www.googleapis.com/auth/service.management.readonly",
      "https://www.googleapis.com/auth/logging.write",
      "https://www.googleapis.com/auth/monitoring.write",
      "https://www.googleapis.com/auth/trace.append",
      "https://www.googleapis.com/auth/devstorage.read_only"
    ]
  }
  network              = "..."
  source_image_family  = "cos-stable"
  source_image_project = "cos-cloud"
  source_image         = reverse(split("/", module.gce-container.source_image))[0]
  labels = {
    "container-vm" = module.gce-container.vm_container_label
  }
}
```

### Terraform Version

```sh
Terraform v1.0.11
on linux_amd64
```

### Additional information

followed: https://github.com/terraform-google-modules/terraform-google-container-vm/tree/master/examples/managed_instance_group

**Comment by T0shik** [comment]
my bad, turns out metadata is quite important here on the mig_template
```
...
source_image         = reverse(split("/", module.gce-container.source_image))[0]
metadata             = { "gce-container-declaration" = module.gce-container.metadata_value }
labels = {
  "container-vm" = module.gce-container.vm_container_label
}
```
